src/queries.jl

```diff
@@ -164,8 +165,8 @@ julia> query(r, Int, 4, <(7))
 
 See also: [`queryall`](@ref), [`tag!`](@ref), [`W`](@ref), [`❓`](@ref)
 """
-function query(reg::Register, tag::Tag, ::Val{allB}=Val{false}(); locked::Union{Nothing,Bool}=nothing, assigned::Union{Nothing,Bool}=nothing, filo::Bool=true) where {allB}
-    _query(reg, tag, Val{allB}(), Val{filo}(); locked=locked, assigned=assigned)
+function query(reg::Register, tag::Tag, ::Val{allB}=Val{false}(); locked::Union{Nothing,Bool}=nothing, assigned::Union{Nothing,Bool}=nothing, filo::Bool=true, ref=nothing) where {allB}
```

Comment:
Didn't we specifically remove `ref` from the public API? Check the new PR that has a lot of extra simplifications and polish to the query functionality as well. There will be a merge conflict and it seems here you only add `iscoherent` -- I would suggest discarding all other changes to this file when handling the merge conflict and just keeping `iscoherent`.